src/main/java/com/twilio/type/RecordingRule.java

```diff
@@ -0,0 +1,195 @@
+package com.twilio.type;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.ToString;
+
+import java.util.Objects;
+
+/**
+ * Recording Rule
+ *
+ * <p>
+ * For more information see:
+ * <a href=https://www.twilio.com/docs/video/api/recording-start/stop#specifying-sr>Specifying Recording Rules</a>
+ * </p>
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@ToString
+public class RecordingRule implements Rule {
+
+    private static final RecordingRule recordAll = builder().withType(Type.INCLUDE).withAll().build();
+    private static final RecordingRule recordNone = builder().withType(Type.EXCLUDE).withAll().build();
+
+    @JsonProperty("type")
+    private final Type type;
+
+    @JsonProperty("all")
+    private final Boolean all;
+
+    @JsonProperty("publisher")
+    private final String publisher;
+
+    @JsonProperty("track")
+    private final String track;
+
+    @JsonProperty("kind")
+    private final Kind kind;
+
+    public RecordingRule(@JsonProperty("type") final Type type,
+            @JsonProperty("all") final Boolean all,
+            @JsonProperty("publisher") final String publisher,
+            @JsonProperty("track") final String track,
+            @JsonProperty("kind") final Kind kind) {
+        this.type = type;
+        this.all = all;
+        this.publisher = publisher;
+        this.track = track;
+        this.kind = kind;
+    }
+
+    public RecordingRule() {
+        this.type = null;
+        this.all = null;
+        this.publisher = null;
+        this.track = null;
+        this.kind = null;
+    }
+
+    public static BuilderStart builder() {
+        return new Builder();
+    }
+
+    public interface BuilderStart {
+        BuilderMiddle withType(final Type type);
+    }
+
+    public interface BuilderMiddle {
+        BuilderMiddleBuild withPublisher(final String publisher);
+        BuilderMiddleBuild withKind(final Kind kind);
+        BuilderMiddleBuild withTrack(final String track);
+        BuilderBuild withAll();
+    }
+
+    public interface BuilderMiddleBuild {
+        BuilderMiddleBuild withPublisher(final String publisher);
+        BuilderMiddleBuild withKind(final Kind kind);
+        BuilderMiddleBuild withTrack(final String track);
+        RecordingRule build();
+    }
```

Comment:
I don't understand the need for these additional interfaces. And one is returning the type of the other.

Reply:
Ah, I see the progression now.